jvm-packages/xgboost4j-spark/src/main/scala/ml/dmlc/xgboost4j/scala/spark/XGBoost.scala

```diff
@@ -304,6 +304,8 @@ object XGBoost extends Serializable {
           " for now")
     }
     require(nWorkers > 0, "you must specify more than 0 workers")
+    require(nWorkers < trainingData.sparkContext.defaultParallelism,
+      "work number greater than defaultParallelism will be hang forever")
```

Comment:
defaultParallelism in fact is the max(spark.default.parallelism, totalCoreCount), and when scheduled, the real parallelism number is min(spark.default.parallelism, totalCoreCount), so I think nWorker < defaultParallelism is a necessary condition but not sufficient.

you're right!